src/main/java/io/jenkins/plugins/conventionalcommits/NextVersionStep.java

```diff
@@ -168,13 +224,36 @@ protected String run() throws Exception {
 
         List<String> commitHistory = Arrays.asList(commitMessagesString.split("\n"));
 
-        // based on the commit list, determine how to bump the version
-        Version nextVersion = new ConventionalCommits().nextVersion(currentVersion, commitHistory);
+        Version nextVersion;
+        if (!incrementPreRelease) {
+          // based on the commit list, determine how to bump the version
+          nextVersion =
+              new ConventionalCommits().nextVersion(currentVersion, commitHistory);
+        } else {
+          nextVersion = currentVersion.incrementPreReleaseVersion();
```

Comment:
Regarding this, pre-release version would be added by 1 irrespective of the commits right?

Reply:
yes, one of my problem is that I wanted to add _preRelease_ option and the derivated options (_incrementPreRelease_ for example) without modify the _ConventionalCommits_ class (by  adding a parameter to the _nextVersion_ method).

The other thing is if we use the commit message to determine if we also increment major / minor / patch version the combinatorial  are quite many.